### PR TITLE
Determine iam endpoint based on token exchange URL in VPC config for …

### PR DESCRIPTION
**Pull request by GunaKKIBM**
…VPC gen2 clusters

Description:

1. Currently, while framing token exchange URL, the env - prod/stage is decided based on the master url present in cluster-info which can cause an issue in dev environments. Moving to deciding stage/prod IAM url based on presence of substring "stage"/"test" in g2_token_exchange_url, would be a better approach.

### CODE REVIEW

**Review by ambiknai — CHANGES_REQUESTED** [review]
Changes lgtm. Please add UT.

**Review by sameshai — APPROVED** [review]
LGTM

**Review by ambiknai — APPROVED** [review]

**Review by prankulmahajan — APPROVED** [review]
lgtm

File: pkg/config/cluster_config.go [hunk 1]
```diff
@@ -101,7 +101,12 @@ func GetTokenExchangeURLfromStorageSecretStore(clusterInfo ClusterConfig, config
 	// Return Private Prod/Stage IAM URL if the cluster is VPC Gen2
 	var providedTokenExchangeURL = false
 	if GetIAASProvider(clusterInfo) == utils.VPCGen2 {
-		if isProduction(clusterInfo.MasterURL) {
+		if isPrivate(config.VPC.G2TokenExchangeURL) {
+			providedTokenExchangeURL = true
+			return config.VPC.G2TokenExchangeURL + tokenExchangePath, providedTokenExchangeURL, nil
```
Comment (GunaKKIBM) [inline pkg/config/cluster_config.go:106]: isURLFramed

Comment (GunaKKIBM) [reply pkg/config/cluster_config.go:106]: Change the naming convention - to make it clear it is. boolean variable

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -71,6 +73,102 @@ func TestParseConfig(t *testing.T) {
...
+		clusterInfoPath          string
+		expectedTokenURL         string
+		providedTokenExchangeURL bool
+		providerToBeUsed         string
+	}{
+		{
+			testCaseName:             "VPC gen2 prod cluster",
+			secretDataPath:           "test-fixtures/valid/vpc-gen2/prod/slclient.toml",
+			clusterInfoPath:          "test-fixtures/valid/vpc-gen2/prod/cluster-info.json",
+			expectedTokenURL:         utils.ProdPrivateIAMURL + tokenExchangePath,
```
Comment (GunaKKIBM) [inline pkg/config/config_test.go:92]: Hard code the return expected value

File: pkg/config/config_test.go [hunk 3]
```diff
@@ -71,6 +73,102 @@ func TestParseConfig(t *testing.T) {
...
+			providedTokenExchangeURL: false,
```
Comment (GunaKKIBM) [inline pkg/config/config_test.go:93]: Change this variable name

File: pkg/config/config_test.go [hunk 4]
```diff
@@ -71,6 +73,102 @@ func TestParseConfig(t *testing.T) {
...
+			providerToBeUsed:         utils.Bluemix,
+		},
+		{
+			testCaseName:             "Satellite cluster prod",
+			secretDataPath:           "test-fixtures/valid/vpc-gen2/prod/satellite/slclient.toml",
+			clusterInfoPath:          "test-fixtures/valid/vpc-gen2/prod/satellite/cluster-info.json",
+			expectedTokenURL:         "https://iam.cloud.ibm.com" + tokenExchangePath,
+			providedTokenExchangeURL: true,
+			providerToBeUsed:         utils.VPC,
+		},
```
Comment (GunaKKIBM) [inline pkg/config/config_test.go:143]: Add couple of more negative test cases

**Review by arahamad — APPROVED** [review]
lgtm

**Pull request merged**